### ISSUE TITLE
accessing kube_config map returned

### ISSUE THREAD

**Issue by mbrancato** [terraform-providers/terraform-provider-azurerm#2988]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

```
Terraform v0.11.10
+ provider.azurerm v1.22.1
+ provider.kubernetes v1.5.2
+ provider.random v1.3.1
+ provider.template v2.1.0
```

### Affected Resource(s)

* `azurerm_kubernetes_cluster`

### Terraform Configuration Files

I put together a module that has a couple different cluster configs, but they look like this:

```hcl
resource "azurerm_kubernetes_cluster" "aks_with_rbac" {
  count               = "${var.enable_rbac == 1 ? 1 : 0}"
  name                = "${var.client_name}-aks-${var.instance_number}"
  location            = "${var.location}"
  resource_group_name = "${var.resource_group_name}"
  dns_prefix          = "${var.client_name}-cluster-${var.instance_number}"
  kubernetes_version  = "${var.kube_version}"

  role_based_access_control {
    enabled = true
    azure_active_directory {

 ...

   }
  }

...

}
```

### Expected Behavior

If the `kube_config` or `kube_admin_config` map is returned, it should be accessible by the calling TF.

### Actual Behavior

The returned data turns into a list of maps, which Terraform has difficulty handling. In order to make it accessible from the calling TF, I had to make the output do this:

```hcl
output "kube_admin_config" {
  description = "AKS Admin Config"
  value       = "${map(
    "host", join("", azurerm_kubernetes_cluster.aks_with_rbac.*.kube_admin_config.0.host),
    "client_certificate", join("", azurerm_kubernetes_cluster.aks_with_rbac.*.kube_admin_config.0.client_certificate),
    "client_key", join("", azurerm_kubernetes_cluster.aks_with_rbac.*.kube_admin_config.0.client_key),
    "cluster_ca_certificate", join("", azurerm_kubernetes_cluster.aks_with_rbac.*.kube_admin_config.0.cluster_ca_certificate)
    )}"
}
```

I believe it should be able to return the map directly, and while it was a list of maps, I was unable to access `kube_admin_config[0]` and `element()` wouldn't work because it complained the list was not flat, even after calling `flatten()`. That said, this could just be a data handling issue on my end, so I'm looking for some guidance here.

### Steps to Reproduce

1. `terraform apply`

**Comment by mbrancato** [comment]
I found out another weird thing. This hack only works when the resource already exists. it doesn't work when trying to create the cluster or an error is returned.

```
Error: Error running plan: 1 error(s) occurred:

* module.k8s_cluster_1.output.kube_admin_config: Resource 'azurerm_kubernetes_cluster.aks_with_rbac' does not have attribute 'kube_admin_config.0.client_key' for variable 'azurerm_kubernetes_cluster.aks_with_rbac.*.kube_admin_config.0.client_key'
```

It does this for all the items in the map.

**Comment by radeksimko** [comment]
Hi @mbrancato 
unfortunately there is no (easy) way to modularize this AKS resource in a way that RBAC block is optional in 0.11. The best way forward in 0.11 is most likely just having two modules, one with RBAC and one without.

However the upcoming `0.12` has some features that make building such dynamic modules possible.

https://gist.github.com/radeksimko/89631735280a496bc6ab8267256c4152

Please note, that while [0.12-beta1 has been released](https://www.hashicorp.com/blog/announcing-terraform-0-1-2-beta1), there's only [a very early version of the AzureRM provider](http://terraform-0.12.0-dev-snapshots.s3-website-us-west-2.amazonaws.com/terraform-provider-azurerm/1.23.0-dev20190216H00-dev/) available (which works with 0.12) for now, while the team is ironing out all bugs/misbehaviours which there may be in that early version.

**Comment by tombuildsstuff** [comment]
hi @mbrancato 

Since this is more of a question for Terraform Core (where this should be solved in the 0.12 release) - rather than keeping this issue open tracking something in another repository I'm going to close this issue for the moment - hopefully the suggestion by @radeksimko to use 2 modules in Terraform 0.11 provides a workaround until Terraform 0.12 becomes available.

Thanks!
